### PR TITLE
Uml 1251 - Send email confirmation on requesting activation key

### PR DESCRIPTION
**Pull request by SeemaMenon**
# Purpose

https://opgtransform.atlassian.net/browse/UML-1251

Fixes UML-1251

## Approach

_Explain how your code addresses the purpose of the change_

## Learning

_Any tips and tricks, blog posts or tools which helped you. Plus anything notable you've discovered about the Use a Lasting Power of Attorney service_

## Checklist

* [X] I have performed a self-review of my own code
* [ ] I have added relevant logging with appropriate levels to my code
* [ ] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [X] I have added tests to prove my work
* [ ] I have added welsh translation tags and updated translation files
* [ ] I have run an accessibility tool on any pages I have made changes to and fixed any issues found
* [ ] The product team have tested these changes

### CODE REVIEW

File: service-front/app/features/actor-add-an-older-lpa.feature [hunk 1]
```diff
@@ -9,12 +9,13 @@ Feature: Add an older LPA
     And I am a user of the lpa application
     And I am currently signed in
 
-  @ui @integration
+#  @ui @integration
```
Comment (cooperaj) [inline service-front/app/features/actor-add-an-older-lpa.feature:12]: Commenting this means the test isn't actually running

File: service-front/app/src/Actor/src/Handler/CheckYourAnswersHandler.php [hunk 2]
```diff
@@ -141,11 +147,21 @@ public function handlePost(ServerRequestInterface $request): ResponseInterface
                         ['user'  => $this->user]
                     ));
                 case OlderLpaApiResponse::SUCCESS:
+
+                    $letterExpectedDate = (new Carbon())->addWeeks(2);
+
+                    $this->emailClient->sendActivationKeyRequestConfirmationEmail(
```
Comment (cooperaj) [inline service-front/app/src/Actor/src/Handler/CheckYourAnswersHandler.php:153]: The sending of letters should be done in the api side. The credentials have now been added but all this should be wrapped up as a part of the code in that service now. 

At some point we'll be removing all notify stuff from the frontend.

File: service-front/app/src/Common/src/Service/Email/EmailClient.php [hunk 3]
```diff
@@ -189,4 +191,33 @@ public function sendSomeoneTriedToUseYourEmailInEmailResetRequest(string $recipi
...
+
+    /**
+     * Send an activation key request confirmation email to a user
+     *
+     * @param string $recipient
+     * @param string $referenceNumber
+     * @param string $postCode
+     * @param string $letterExpectedDate
+     */
+    public function sendActivationKeyRequestConfirmationEmail(
```
Comment (cooperaj) [inline service-front/app/src/Common/src/Service/Email/EmailClient.php:203]: This class needs copying into the api service and alterations made there.

**Review by cooperaj — CHANGES_REQUESTED** [review]

File: service-front/app/src/Actor/src/Handler/CheckYourAnswersHandler.php (hunk 2 above)
Comment (SeemaMenon) [reply service-front/app/src/Actor/src/Handler/CheckYourAnswersHandler.php:153]: Created new improvement ticket for the logic shift exercise: 
https://opgtransform.atlassian.net/browse/UML-1315

File: service-front/app/src/Common/src/Service/Email/EmailClient.php (hunk 3 above)
Comment (SeemaMenon) [reply service-front/app/src/Common/src/Service/Email/EmailClient.php:203]: Created new improvement ticket for the logic shift exercise:
https://opgtransform.atlassian.net/browse/UML-1315

File: service-front/app/features/actor-add-an-older-lpa.feature (hunk 1 above)
Comment (SeemaMenon) [reply service-front/app/features/actor-add-an-older-lpa.feature:12]: Oh..that's a mistake! Sorry...

File: service-front/app/src/Common/src/Service/Email/EmailClient.php [hunk 4]
```diff
@@ -189,4 +191,33 @@ public function sendSomeoneTriedToUseYourEmailInEmailResetRequest(string $recipi
...
+        string $recipient,
+        string $referenceNumber,
+        string $postCode,
+        string $letterExpectedDate
+    ){
+        if ($this->locale === "cy") {
+            $this->notifyClient->sendEmail($recipient, self::WELSH_TEMPLATE_ID_ACTIVATION_KEY_REQUEST_CONFIRMATION, [
+                'reference_number' => $referenceNumber,
```
Comment (GemTay) [inline service-front/app/src/Common/src/Service/Email/EmailClient.php:211]: in notify, the reference number attribute is written as "ref", so i dont know if this wouldnt work?

**Review by GemTay — CHANGES_REQUESTED** [review]

File: service-front/app/src/Common/src/Service/Email/EmailClient.php (hunk 4 above)
Comment (SeemaMenon) [reply service-front/app/src/Common/src/Service/Email/EmailClient.php:211]: The actual template is not yet created I guess. Uchenna has just created placeholders and passed me the id's for usage in code. Sarah is aware and hence UAT pending.

Comment (SeemaMenon) [reply service-front/app/src/Common/src/Service/Email/EmailClient.php:211]: I don't have access to notify ATM, not able to receive the text message code.

Comment (GemTay) [reply service-front/app/src/Common/src/Service/Email/EmailClient.php:211]: The templates are created, thats what the template ID represents, so just change the attribute name to 'ref' so it's assigned the value correctly in the templates

Comment (SeemaMenon) [reply service-front/app/src/Common/src/Service/Email/EmailClient.php:211]: @GemTay  -  Got you on this. Thanks for pointing this out.

@andrewpearce-digital kindly updated the reference on the templates. The functionality works now on testing.

Request a review again please.

**Review by GemTay — APPROVED** [review]

**Review by cooperaj — APPROVED** [review]

**Pull request merged**